tdrs-backend/tdpservice/parsers/schema_defs/tanf/t6.py

```diff
@@ -0,0 +1,168 @@
+"""Schema for HEADER row of all submission types."""
+
+
+from ...util import SchemaManager
+from ...fields import Field
+from ...row_schema import RowSchema
+from ... import validators
+from tdpservice.search_indexes.models.tanf import TANF_T6
+
+
+s1 = RowSchema(
+    model=TANF_T6,
+    preparsing_validators=[
+        validators.hasLength(379),
+    ],
+    postparsing_validators=[
+        validators.sumIsEqual("NUM_APPLICATIONS", ["NUM_APPROVED", "NUM_DENIED"]),
+        validators.sumIsEqual("NUM_FAMILIES", ["NUM_2_PARENTS", "NUM_1_PARENTS", "NUM_NO_PARENTS"]),
+        validators.sumIsEqual("NUM_RECIPIENTS", ["NUM_ADULT_RECIPIENTS", "NUM_CHILD_RECIPIENTS"]),
+    ],
+    fields=[
+        Field(item="0", name='RecordType', type='string', startIndex=0, endIndex=2,
+              required=True, validators=[]),
+        Field(item="3A", name='CALENDAR_YEAR', type='number', startIndex=2, endIndex=6,
+              required=True, validators=[validators.month_year_yearIsLargerThan(1998)]),
+        Field(item="3B", name='CALENDAR_QUARTER', type='number', startIndex=6, endIndex=7,
+              required=True, validators=[validators.isInLimits(1, 4)]),
+        Field(item="4A", name='NUM_APPLICATIONS', type='number', startIndex=7, endIndex=15,
+              required=True, validators=[validators.isInLimits(0, 99999999)]),
```

Comment:
While this might be technically permissible to have all these 9**** field validations, is there a cleaner solution? I just don't love seeing all these hard-coded and sort of absurd "magic number" values. Perhaps we just derive length and push that down to `isInLimits()`? Like, common sense says to me there won't be 99 million applications per quarter or esp. the assistance value being so high...